rotkehlchen/tests/unit/decoders/test_sdai_gnosis.py

```diff
@@ -0,0 +1,245 @@
+    tx_hash = deserialize_evm_tx_hash('0x1342646cab122d58f0b7dfae404dad5235d42224de881099dc05e59477bb93aa')  # noqa: E501
+    actual_events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=gnosis_inquirer,
+        database=database,
+        tx_hash=tx_hash
+    )
+    location_label = b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00x\xe8wW\x86\x11\x85\xec^\x8c\x0e\xf6\xbf\x0ci\xfax2\xdfl'
+    timestamp = TimestampMS(1707169525000)
+    gas_amount = '0.000367251244452481'
+    deposit_amount = '315'
+    withdraw_amount = '303.052244055946806232'
+    expected_events = [
+        EvmEvent(sequence_index=0,
+                 timestamp=timestamp,
+                 location=Location.GNOSIS,
+                 event_type=HistoryEventType.SPEND,
+                 event_subtype=HistoryEventSubType.FEE,
+                 asset=A_XDAI,
+                 balance=Balance(amount=FVal(gas_amount)),
+                 location_label=user_address,
+                 notes=f'Burned {gas_amount} XDAI for gas',
+                 tx_hash=tx_hash,
+                 counterparty=CPT_GAS,
+                 ),
+        EvmEvent(sequence_index=3021,
+                 timestamp=timestamp,
+                 location=Location.GNOSIS,
+                 event_type=HistoryEventType.DEPOSIT,
+                 event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+                 asset=A_XDAI,
+                 balance=Balance(amount=FVal(deposit_amount)),
+                 location_label=user_address,
+                 notes=f'Deposit {deposit_amount} DAI to sDAI contract',
+                 tx_hash=tx_hash,
+                 counterparty=CPT_SDAI,
+                 address='0x83F20F44975D03b1b09e64809B757c47f942BEeA',
+                 ),
+        EvmEvent(sequence_index=3022,
+                 timestamp=timestamp,
+                 location=Location.GNOSIS,
+                 event_type=HistoryEventType.WITHDRAWAL,
+                 event_subtype=HistoryEventSubType.REMOVE_ASSET,
+                 asset=A_SDAI,
+                 balance=Balance(amount=FVal(withdraw_amount)),
+                 location_label=location_label,
+                 notes=f'Withdraw {withdraw_amount} sDAI from sDAI contract',
+                 tx_hash=tx_hash,
+                 counterparty=CPT_SDAI,
+                 address='0x83F20F44975D03b1b09e64809B757c47f942BEeA',
+                 ),
+        EvmEvent(sequence_index=3023,
+                 timestamp=timestamp,
+                 location=Location.GNOSIS,
+                 event_type=HistoryEventType.WITHDRAWAL,
+                 event_subtype=HistoryEventSubType.REMOVE_ASSET,
+                 asset=Asset('eip155:100/erc20:0xaf204776c7245bF4147c2612BF6e5972Ee483701'),
+                 balance=Balance(amount=FVal(withdraw_amount)),
+                 location_label=user_address,
+                 notes=f'Withdraw {withdraw_amount} DAI from sDAI contract',
+                 tx_hash=tx_hash,
+                 counterparty=CPT_SDAI,
```

Comment:
this looks wrong. You should have 3 events here. The gas event, the deposit of DAI and the withdrawal of sDAI. Here you have one extra withdrawal of DAI that shouldn't exist